woocommerce/woocommerce
# [CYS - Core] Improve loading screen

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR improves the loading screen. Given that `Loader` component is a shared component, I just extract the logic for create the steps, but I kept two pages for the two flows.

Closes #43456.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new WooCommerce installation with this version.
2. Ensure the WooCommerce Beta Tester plugin is installed and activated (available on this monorepo).
3. Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag.
4. Visit the `wp-admin/admin.php?page=wc-admin&path=/customize-store`
5. Follow the process.
6. Ensure that the loading page screen matches the video:


https://github.com/woocommerce/woocommerce/assets/4463174/76b0bc16-cef8-46cb-b644-4f11db8e2b4f






### Changelog entry




<details>

#### Significance




#### Type




#### Message 

[CYS - Core] Improve loading screen.

#### Comment 

</details>
